### PR TITLE
Get Spree Reviews Working on Spree 4.1

### PR DESCRIPTION
**Pull request by MatthewKennedy**
- Updated to work on Spree 4.1 and up, including Spree 4.2 running on Ruby 3.0
- Added a few UI improvements. (link to view all reviews & a link back by cancel writing a review)
- Added pagination to reviews index page.
- Linted the ERB and CSS.

Details are listed in the change log.

### CODE REVIEW

**Review by poudelprakash — APPROVED** [review]

**Comment by poudelprakash** [comment]
@damianlegawiec we have been waiting for spree 4 support for this gem, Hope you will have time to review this :)

**Review by AhmedAliIbrahim — APPROVED** [review]

File: spree_reviews.gemspec [hunk 1]
```diff
@@ -9,38 +9,26 @@ Gem::Specification.new do |s|
...
-  s.add_runtime_dependency 'spree_core', spree_version
-  s.add_runtime_dependency 'spree_auth_devise', spree_version
-  s.add_runtime_dependency 'spree_extension'
+  spree_version = '>= 3.2.0', '< 5.0'
+  s.add_dependency 'spree_api', spree_version
+  s.add_dependency 'spree_backend', spree_version
+  s.add_dependency 'spree_core', spree_version
+
+  s.add_dependency 'deface', '~> 1.0'
+  s.add_dependency 'spree_auth_devise'
```
Comment (damianlegawiec) [inline spree_reviews.gemspec:29]: @MatthewKennedy I think we can skip spree auth devise here, it's only for running test etc

Comment (MatthewKennedy) [reply spree_reviews.gemspec]: Sorted.

File: app/views/spree/reviews/_form.html.erb [hunk 2]
```diff
@@ -31,6 +31,8 @@
   <% end %>
 
   <div class="form-actions">
-    <%= f.submit Spree.t(:submit_your_review), class: 'btn btn-default' %>
+    <%= f.submit Spree.t(:submit_your_review), class: 'btn btn-primary' %>
+         <%= Spree.t(:or) %>
+    <%= link_to Spree.t(:cancle), product_path(@product), class: 'btn btn-outline-primary', rel: 'nofollow' %>
```
Comment (damianlegawiec) [inline app/views/spree/reviews/_form.html.erb:36]:
```suggestion
    <%= link_to Spree.t(:cancel), product_path(@product), class: 'btn btn-outline-primary', rel: 'nofollow' %>
```

**Review by damianlegawiec — APPROVED** [review]
small typo to fix and ready to go!

awesome work 💯

File: app/views/spree/reviews/_form.html.erb (hunk 2 above)
Comment (MatthewKennedy) [reply app/views/spree/reviews/_form.html.erb]: Will sort that.

**Pull request merged**